### PR TITLE
 teacher api

### PR DESCRIPTION
**Pull request by david-ron**

### CODE REVIEW

**Review by leojrsilao — APPROVED** [review]

File: src/main/java/com/kimsquitsystem/kimsquitdemo/dao/TeacherDao.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    public List<Teacher> selectAllTeachers() {
+        final String sql = "SELECT teacher_id, first_name, last_name FROM teacher";
+        List<Teacher> teachers = jdbcTemplate.query(sql, (resultSet, i) ->
+        {
+            int user_id = resultSet.getInt("teacher_id");
+            String firstName = resultSet.getString("first_name");
+            String lastName = resultSet.getString("last_name");
+            return new Teacher(user_id, firstName,lastName);
```
Comment (bit172) [inline src/main/java/com/kimsquitsystem/kimsquitdemo/dao/TeacherDao.java]: Add the address too

File: src/main/java/com/kimsquitsystem/kimsquitdemo/dao/TeacherDao.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    public Optional<Teacher> selectByTeacherId(String id) {
+        final String sql = "SELECT teacher_id, first_name, last_name FROM teacher WHERE teacher_id = ?";
+        Teacher teacher = jdbcTemplate.queryForObject(sql, new Object[]{ id }, (resultSet, i) ->
+        {
+            int user_id = resultSet.getInt("teacher_id");
+            String firstName = resultSet.getString("first_name");
+            String lastName = resultSet.getString("last_name");
+            return new Teacher(user_id, firstName, lastName);
```
Comment (bit172) [inline src/main/java/com/kimsquitsystem/kimsquitdemo/dao/TeacherDao.java]: Here too

**Review by bit172 — CHANGES_REQUESTED** [review]

File: src/main/java/com/kimsquitsystem/kimsquitdemo/dao/TeacherDao.java (hunk 2 above)
Comment (david-ron) [reply src/main/java/com/kimsquitsystem/kimsquitdemo/dao/TeacherDao.java]: should b good now

File: src/main/java/com/kimsquitsystem/kimsquitdemo/dao/TeacherDao.java (hunk 1 above)
Comment (david-ron) [reply src/main/java/com/kimsquitsystem/kimsquitdemo/dao/TeacherDao.java]: should b good now

**Review by bit172 — APPROVED** [review]

**Pull request merged**